File: java/src/main/java/com/lacombe/promo3/taxi/TaxiOrganizer.java
package com.lacombe.promo3.taxi;

import java.time.LocalTime;

public class TaxiOrganizer {

    private final Arrivals arrivals;

    public TaxiOrganizer(Arrivals arrivals) {
        this.arrivals = arrivals;
    }

    public TaxiBooking getBookings() {

        final ArrivalHour lastArrivalTime = arrivals.getLastArrivalTime();

        Taxi taxi = new Taxi("Taxi_"+ lastArrivalTime +"h");
        LocalTime departureTime = lastArrivalTime.getArrivalTime();
        Passengers passengers = new Passengers();

        for (Arrival arrival : arrivals.getArrivals()) {
            passengers.add(new Passenger(arrival.getParticipantName()));
        }

        return new TaxiBooking(taxi, departureTime, passengers);
    }

}

File: dotnet/Socrates/TaxiOrganization/TaxiOrganizer.cs
using System.Collections.Generic;
using System.Linq;

namespace Socrates.TaxiOrganization
{
    public class TaxiOrganizer
    {
        private readonly Arrivals arrivals;

        public TaxiOrganizer(Arrivals arrivals)
        {
            this.arrivals = arrivals;
        }

        public TaxiBooking GetBookings()
        {
            var taxi = new Taxi($"Taxi_{ arrivals.GetLastArrivalTime() }h");

            var time = arrivals.GetLastArrivalTime().ToString();

            var passengers = arrivals.GetArrivals().Select(arrival => new Passenger(arrival.GetParticipantName())).ToList();

            return new TaxiBooking(taxi, time, passengers);
        }

    }
}
